modules/hocon/src/main/scala/ciris/hocon/package.scala

```diff
@@ -0,0 +1,33 @@
+package ciris
+
+import cats.Show
+import ciris.{ConfigException => _, _}
+import com.typesafe.{config => tsc}
+import cats.syntax.all
+
+import scala.util.Try
+
+package object hocon extends HoconConfigDecoders {
+
+  final class HoconAt(configV: ConfigValue[tsc.Config]) {
+
+    def apply(name: String): ConfigValue[tsc.ConfigValue] =
+      ConfigValue.suspend {
+        val key = ConfigKey(s"hocon key $name")
+        configV.flatMap { config =>
```

Comment:
To be honest I would prefer no to impose `flatMap`on to the user for him to be able to use api exposed by `HoconAt` (btw. I will rename this class, it doesnt actually make sense). Wouldnt it be enough to fix the loading as you advised one comment earlier?

Reply:
To be honest I would like to avoid forcing users to `flatMap` only to get access to the api. But then if we want to cache the `Config` on creation then the creation becomes effectful and must be contained ... 

At one moment I had an alternative design where `Config` was passed by the user into the hocon support. I resigned from this approach because then user has to write `IO(ConfigFactory.load)` which doesnt help either... For now I dont see other options but I will give some more thought. Maybe some `unsafe` method as in log4cats or in monix `memoize`? Its not that uncommon to opt-out of safety for allocating mutable state

Reply:
Well, `flatMap` is really _the_ way to share state when using effect types, so you can't really get around it. It's fine if we have functions prefixed with `unsafe`, but we still need a safe version with `flatMap` though.